### ISSUE TITLE
Ejabberd doesn't pass in log in request to external authentication script

### ISSUE THREAD

**Issue by jcho7022** [processone/ejabberd#2423]
> What version of ejabberd are you using?

Lastest Ejabberd docker image

> What operating system (version) are you using?

Docker (not sure what os it is)

> How did you install ejabberd (source, package, distribution)?

Installed using a docker image

> What did not work as expected? Are there error messages in the log? 

Python script does not receive a input when sending a login request using android smack, however, oddly when sending a login request using Adium Ejabberd passes in the input to the script. I think something goes wrong when decodeing the plain encrypted request or possibly when the smack libray encrypts it.

Also, when using ejabberd version 16.04 Ejabberd passes in the request without any problem however 18.xx ejabberd does not.

This is the debug logs when requesting a login from my android device using a smack library
```
1. Accepted connection :: # some IP

2. Received XML on stream = <<"<stream:stream xmlns='jabber:client' to=examp.com' xmlns:stream='http://etherx.jabber.org/streams' version='1.0' xml:lang='en'>">>

3. Send XML on stream = <<"<?xml version='1.0'?><stream:stream id='#number' version='1.0' xml:lang='en' xmlns:stream='http://etherx.jabber.org/streams' from='examp.com' xmlns='jabber:client'>">>

4. Send XML on stream = <<"<stream:features><mechanisms xmlns='urn:ietf:params:xml:ns:xmpp-sasl'><mechanism>PLAIN</mechanism></mechanisms></stream:features>">>

5 .Received XML on stream = <<"<auth xmlns='urn:ietf:params:xml:ns:xmpp-sasl' mechanism='PLAIN'>AEphbWVzAGV5SmhiR2NpT2lKU1V6STFOaUlzSW5SNWNDSTZJa3BYVkNKOS5leUpwWVhRaU9qRTFNalkzT0RVMk56UXNJbVY0Y0NJNk1UVXlOamM0TnpRM05Dd2lhWE56SWpvaVRXRndRWEJ3SWl3aWMzVmlJam9pU21GdFpYTWlmUS5leENwS25Na0J5WldobnBfbzFwSDN6R2tHTGNEdjZSZTVIbjZRV3NMeURuS0VGT2ZoZGlzajY0S0pXRDRWUVF5WmJrejNvcGRfaTU5TXE3WlY4WmhkRFF1a2xfUjFCMmdXYl80S2hOVmk3dW5ublFhWTBMUTZXQ0R4OGhUbEQwOTgybjRKVEJndW1UbGt4TFY3U1FkSkt0ejhyMWFQaUpIcmdQOFJsM2ttVUN3N3dYaVRCVS1rY0dUcWEzcDZsa0pQT1lFMWlHYkN2UVFZWUU1eEI2Y3RhUzZ6dGRhcm4yZDlFS2FacXprNkhubExBZnctV0R5OThCVnc0cjRwR3RpNWo1U2RPQ21GUGxPdnROUnZxTnFRV2pycW9QOGsxLTRIUGRaMTh2RlFHelBlUFczZW96Slpva1lGQ2NoTFI3WEh3LTI3SzBMNjJ3M3NBSUg5S1R4U3c=</auth>">>

6. State: {maxrate,1000,0.0,1526785680539806}, Size=704
------------------------------------------------------------------------------------
DOES NOT PASS THE REQUEST TO MY EXTERNAL AUTHENTICATION SCRIPT
------------------------------------------------------------------------------------
7. Failed c2s PLAIN authentication for xxx@examp.com from ::FFFF:128.189.139.25: Invalid username or password . 
^^^^^^ without receiving a response from the script reponds back 
```

Debug logs when requesting login from Adium.
```
1. Accepted connection :: # some IP

2. Received XML on stream = <<"<?xml version='1.0' ?>">>

3. Received XML on stream = <<"<stream:stream to='examp.com' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' version='1.0'>">>

4. 03:42:10.523 [debug] (tcp|<0.1039.0>) Send XML on stream = <<"<?xml version='1.0'?><stream:stream id='6042112340145509377' version='1.0' xml:lang='en' xmlns:stream='http://etherx.jabber.org/streams' from='ex.com' xmlns='jabber:client'>">>

5. Send XML on stream = <<"<stream:features><mechanisms xmlns='urn:ietf:params:xml:ns:xmpp-sasl'><mechanism>PLAIN</mechanism></mechanisms></stream:features>">>

6. Received XML on stream = <<"<auth xmlns='urn:ietf:params:xml:ns:xmpp-sasl' mechanism='PLAIN' xmlns:ga='http://www.google.com/talk/protocol/auth' ga:client-uses-full-bind-result='true'>AGphbWVzAGV5SmhiR2NpT2lKU1V6STFOaUlzSW5SNWNDSTZJa3BYVkNKOS5leUpwWVhRaU9qRTFNalkzTlRrek9ETXNJbVY0Y0NJNk1UVXlOamMyTVRFNE15d2lhWE56SWpvaVRXRndRWEJ3SWl3aWMzVmlJam9pU21GdFpYTWlmUS5rTGRMaWV6NUJickg5bThTX0ZmSG42Z1RnTUVXUWVPald1SDEwaFc2UGo0OVN4VGE5T0paekEtVXRPTWE3NHVIRXc5Ym44WUV6bFNWZXdHeFJpM21ZZVp3Sk9kQVBGdkZsN1NCT21KcnhHazlrMk5JSmYtUkdXV3pBLWduTURaaE90Vi16a0lVYUkxamZsQmVsS1NxT2gtdHRxM2hkUF9WT3MyOFJJQm9KU3JMUkQtQ3dEYVptaG5WRHFQRm1pZ19aX3NiVHI0RzZ5UHhhM0RWeUNfNi1YX3k0TU05a1hkeXFhMkZWeHZJOXN5b1BKMGRPVEJkN28za21Oa21SUHBnUFZwblV2VXFQQTRQTkV4Q1UzbjdDV2dncUlTejNDX21kaG9qM1FhdHJtamlmcTBYQVVBSE9RWG1taHVYeTZkOG0wU0p2bHNfQVlwTm8xano2NjJBZkE=</auth>">>

7.  State: {maxrate,1000,0.0,1526787730523478}, Size=795

8. extauth call '[<<"auth">>,<<"james">>,<<"ec2-34-217-26-152.us-west-2.compute.amazonaws.com">>,<<"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1MjY3NTkzODMsImV4cCI6MTUyNjc2MTE4MywiaXNzIjoiTWFwQXBwIiwic3ViIjoiSmFtZXMifQ.kLdLiez5BbrH9m8S_FfHn6gTgMEWQeOjWuH10hW6Pj49SxTa9OJZzA-UtOMa74uHEw9bn8YEzlSVewGxRi3mYeZwJOdAPFvFl7SBOmJrxGk9k2NIJf-RGWWzA-gnMDZhOtV-zkIUaI1jflBelKSqOh-ttq3hdP_VOs28RIBoJSrLRD-CwDaZmhnVDqPFmig_Z_sbTr4G6yPxa3DVyC_6-X_y4MM9kXdyqa2FVxvI9syoPJ0dOTBd7o3kmNkmRPpgPVpnUvUqPA4PNExCU3n7CWggqISz3C_mdhoj3Qatrmjifq0XAUAHOQXmmhuXy6d8m0SJvls_AYpNo1jz662AfA">>]

9. Send XML on stream = <<"<stream:error><connection-timeout xmlns='urn:ietf:params:xml:ns:xmpp-streams'/><text xml:lang='en' xmlns='urn:ietf:params:xml:ns:xmpp-streams'>Idle connection</text></stream:error>">>

10. Send XML on stream = <<"</stream:stream>">>
```
  

My Dockerfile
```FROM rroemhild/ejabberd

WORKDIR /usr/src/app

USER root

RUN apt-get update && apt-get install -y python3 build-essential libssl-dev libffi-dev python-dev &&\
    apt-get install -y python-pip  && \
    pip install PyJWT &&\
    pip install cryptography


ADD ./ejabberd.yml /opt/ejabberd/conf/ejabberd.yml.tpl
ADD ./ejabberdctl.cfg /opt/ejabberd/conf/ejabberdctl.cfg.tpl
ADD ./extauth.log /var/log/ejabberd/extauth.log

COPY . .

USER ejabberd

EXPOSE 5222
EXPOSE 5269
EXPOSE 5280
```

External authentication python script
```
import jwt
import string
import MySQLdb, logging
import sys
import struct
import os

db = None;

logging.basicConfig(level=logging.INFO,
                    format='%(asctime)s %(levelname)s %(message)s',
                    filename='/var/log/ejabberd/extauth.log',
                    filemode='a')

class DB:
    conn = None;

    def connect(self):
        self.conn =  MySQLdb.connect( host = '127.0.0.1', user = 'xxx', passwd = 'xxx', db = 'xxx', port = xxx)

    def query(self,sql):
        try:
            cursor = self.conn.cursor()
            cursor.execute(sql)
        except (AttributeError, MySQLdb.OperationalError ):
            self.connect()
            cursor = self.conn.cursor()
            cursor.execute(sql)
        return cursor


def from_ejabberd():
    logging.info("Getting something from ejabberd")
    input_length = sys.stdin.read(2)
---------------------------------------- PROGRAM STOPS BEFORE THIS LINE AND WAITS FOR A RESPONSE BUT NEVER RECIEVES ONE 
    logging.info("Bytes read: " + str(len(input_length)))
    logging.info("Input Length: " + str(input_length))
    (size,) = struct.unpack('>h', input_length)
    return sys.stdin.read(size).split(':')

def to_ejabberd(bool):
    answer = 0
    if bool:
        answer = 1
    token = struct.pack('>hh', 2, answer)
    sys.stdout.write(token)
    sys.stdout.flush()

def auth(username, server, password):
    pemfile = open("/usr/src/app/auth/jwtrsa-public.pem", 'r')
    keystring = pemfile.read()
    logging.info('Token: ' + password +'keystring:' + keystring)
    payload = jwt.decode(password, keystring, verify=True, algorithms=['RS256'])
    key = 'xxx'
    logging.info( payload)
    logging.info('Token payload: ' + payload[key] +'username:' + username)

    if payload[key].lower() == username.lower() and isuser(username, "") :
        return True
    else:
        return False

def isuser(username, server):
    sql = "xxxx"
    cursor = db.query(sql)
    cursor.fetchall()
    valid = cursor.rowcount
    logging.info(valid)
    if valid == 1:
        return True
    else:
        return False


def setpass(username, server, password):
    return False

db = DB()
logging.info('Program has started')
while True:
    logging.info('program is running')
    data = from_ejabberd()
    success = False
    logging.info('program is running got rec')
    if data[0] == "auth":
        logging.info('authentication request')
        success = auth(data[1], data[2], data[3])
    elif data[0] == "isuser":
        success = isuser(data[1], data[2])
    elif data[0] == "setpass":
        success = setpass(data[1], data[2], data[3])
    to_ejabberd(success)

logging.info('Program is ending')
db.close();

```

My ejabberd.yml file
```
###
###'              ejabberd configuration file
###
###

### The parameters used in this configuration file are explained in more detail
### in the ejabberd Installation and Operation Guide.
### Please consult the Guide in case of doubts, it is included with
### your copy of ejabberd, and is also available online at
### http://www.process-one.net/en/ejabberd/docs/

### The configuration file is written in YAML.
### Refer to http://en.wikipedia.org/wiki/YAML for the brief description.
### However, ejabberd treats different literals as different types:
###
### - unquoted or single-quoted strings. They are called "atoms".
###   Example: dog, 'Jupiter', '3.14159', YELLOW
###
### - numeric literals. Example: 3, -45.0, .0
###
### - quoted or folded strings.
###   Examples of quoted string: "Lizzard", "orange".
###   Example of folded string:
###   > Art thou not Romeo,
###     and a Montague?

###.  =======
###'  LOGGING

##
## loglevel: Verbosity of log files generated by ejabberd.
## 0: No ejabberd log at all (not recommended)
## 1: Critical
## 2: Error
## 3: Warning
## 4: Info
## 5: Debug
##
loglevel: 5

##
## rotation: Describe how to rotate logs. Either size and/or date can trigger
## log rotation. Setting count to N keeps N rotated logs. Setting count to 0
## does not disable rotation, it instead rotates the file and keeps no previous
## versions around. Setting size to X rotate log when it reaches X bytes.
## To disable rotation set the size to 0 and the date to ""
## Date syntax is taken from the syntax newsyslog uses in newsyslog.conf.
## Some examples:
##  $D0     rotate every night at midnight
##  $D23    rotate every day at 23:00 hr
##  $W0D23  rotate every week on Sunday at 23:00 hr
##  $W5D16  rotate every week on Friday at 16:00 hr
##  $M1D0   rotate on the first day of every month at midnight
##  $M5D6   rotate on every 5th day of the month at 6:00 hr
##
log_rotate_size: 10485760
log_rotate_date: ""
log_rotate_count: 1

##
## overload protection: If you want to limit the number of messages per second
## allowed from error_logger, which is a good idea if you want to avoid a flood
## of messages when system is overloaded, you can set a limit.
## 100 is ejabberd's default.
log_rate_limit: 100

##
## watchdog_admins: Only useful for developers: if an ejabberd process
## consumes a lot of memory, send live notifications to these XMPP
## accounts.
##
## watchdog_admins:
##   - "bob@example.com"

###.  ===============
###'  NODE PARAMETERS

##
## net_ticktime: Specifies net_kernel tick time in seconds. This options must have
## identical value on all nodes, and in most cases shouldn't be changed at all from
## default value.
##
## net_ticktime: 60

###.  ================
###'  SERVED HOSTNAMES

##
## hosts: Domains served by ejabberd.
## You can define one or several, for example:
## hosts:
##   - "example.net"
##   - "example.com"
##   - "example.org"
##
hosts:
  - "xxxx"

##
## route_subdomains: Delegate subdomains to other XMPP servers.
## For example, if this ejabberd serves example.org and you want
## to allow communication with an XMPP server called im.example.org.
##
## route_subdomains: s2s

###.  ============
###'  Certificates

## List all available PEM files containing certificates for your domains,
## chains of certificates or certificate keys. Full chains will be built
## automatically by ejabberd.
##
## certfiles:
##   - "/etc/letsencrypt/live/example.org/*.pem"
##   - "/etc/letsencrypt/live/example.com/*.pem"
##
## If your system provides only a single CA file (CentOS/FreeBSD):
## ca_file: "/etc/ssl/certs/ca-bundle.pem"

###.  =================
###'  TLS configuration

## Note that the following configuration is the default
## configuration of the TLS driver, so you don't need to
## uncomment it.
##
## define_macro:
##   'TLS_CIPHERS': "HIGH:!aNULL:!eNULL:!3DES:@STRENGTH"
##   'TLS_OPTIONS':
##     - "no_sslv3"
##     - "cipher_server_preference"
##     - "no_compression"
##   'DH_FILE': "/path/to/dhparams.pem" # generated with: openssl dhparam -out dhparams.pem 2048
##
## c2s_dhfile: 'DH_FILE'
## s2s_dhfile: 'DH_FILE'
## c2s_ciphers: 'TLS_CIPHERS'
## s2s_ciphers: 'TLS_CIPHERS'
## c2s_protocol_options: 'TLS_OPTIONS'
## s2s_protocol_options: 'TLS_OPTIONS'

###.  ===============
###'  LISTENING PORTS

##
## listen: The ports ejabberd will listen on, which service each is handled
## by and what options to start it with.
##
listen:
  -
    port: 5222
    ip: "::"
    module: ejabberd_c2s
    ##
    ## If TLS is compiled in and you installed a SSL
    ## certificate, uncomment this line:
    ##
    ## starttls: true
    ##
    ## To enforce TLS encryption for client connections,
    ## use this instead of the "starttls" option:
    ##
    ## starttls_required: true
    ##
    ## Stream compression
    ##
    ## zlib: true
    ##
    max_stanza_size: 65536
    shaper: c2s_shaper
    access: c2s
    stream_management: true
    resume_timeout: 5
    resend_on_timeout: true
    

  -
    port: 5269
    ip: "::"
    module: ejabberd_s2s_in
  -
    port: 5280
    ip: "::"
    module: ejabberd_http
    request_handlers:
      "/ws": ejabberd_http_ws
      "/bosh": mod_bosh
      "/api": mod_http_api
    ##  "/pub/archive": mod_http_fileserver
    web_admin: true
    ## register: true
    captcha: true
  ##
  ## ejabberd_service: Interact with external components (transports, ...)
  ##
  ## -
  ##   port: 8888
  ##   ip: "::"
  ##   module: ejabberd_service
  ##   access: all
  ##   shaper_rule: fast
  ##   ip: "127.0.0.1"
  ##   privilege_access:
  ##      roster: "both"
  ##      message: "outgoing"
  ##      presence: "roster"
  ##   delegations:
  ##      "urn:xmpp:mam:1":
  ##        filtering: ["node"]
  ##      "http://jabber.org/protocol/pubsub":
  ##        filtering: []
  ##   hosts:
  ##     "icq.example.org":
  ##       password: "secret"
  ##     "sms.example.org":
  ##       password: "secret"

  ##
  ## ejabberd_stun: Handles STUN Binding requests
  ##
  ## -
  ##   port: 3478
  ##   transport: udp
  ##   module: ejabberd_stun

  ##
  ## To handle XML-RPC requests that provide admin credentials:
  ##
  ## -
  ##   port: 4560
  ##   ip: "::"
  ##   module: ejabberd_xmlrpc
  ##   maxsessions: 10
  ##   timeout: 5000
  ##   access_commands:
  ##     admin:
  ##       commands: all
  ##       options: []

  ##
  ## To enable secure http upload
  ##
  ## -
  ##   port: 5444
  ##   ip: "::"
  ##   module: ejabberd_http
  ##   request_handlers:
  ##     "": mod_http_upload
  ##   tls: true
  ##   protocol_options: 'TLSOPTS'
  ##   dhfile: 'DHFILE'
  ##   ciphers: 'CIPHERS'

## Disabling digest-md5 SASL authentication. digest-md5 requires plain-text
## password storage (see auth_password_format option).
disable_sasl_mechanisms: 
  - "digest-md5"
  - "X-OAUTH2"
  - "SCRAM-SHA-1"

###.  ==================
###'  S2S GLOBAL OPTIONS

##
## s2s_use_starttls: Enable STARTTLS for S2S connections.
## Allowed values are: false, optional or required
## You must specify 'certfiles' option
##
## s2s_use_starttls: optional

##
## S2S whitelist or blacklist
##
## Default s2s policy for undefined hosts.
##
## s2s_access: s2s

##
## Outgoing S2S options
##
## Preferred address families (which to try first) and connect timeout
## in seconds.
##
## outgoing_s2s_families:
##   - ipv4
##   - ipv6
## outgoing_s2s_timeout: 190

###.  ==============
###'  AUTHENTICATION

##
## auth_method: Method used to authenticate the users.
## The default method is the internal.
## If you want to use a different method,
## comment this line and enable the correct ones.
##

##
## Store the plain passwords or hashed for SCRAM:
##auth_password_format: plain
##auth_password_format: scram
##
## Define the FQDN if ejabberd doesn't detect it:
## fqdn: "server3.example.com"

##
## Authentication using external script
## Make sure the script is executable by ejabberd.
##
auth_method: [external]
extauth_program: "/usr/src/app/auth/exAuth.py"
extauth_instances: 1
auth_use_cache: false

##
## Authentication using SQL
## Remember to setup a database in the next section.
##
##auth_method: sql

##
## Authentication using PAM
##
## auth_method: pam
## pam_service: "pamservicename"

##
## Authentication using LDAP
##
## auth_method: ldap
##
## List of LDAP servers:
## ldap_servers:
##   - "localhost"
##
## Encryption of connection to LDAP servers:
## ldap_encrypt: none
## ldap_encrypt: tls
##
## Port to connect to on LDAP servers:
## ldap_port: 389
## ldap_port: 636
##
## LDAP manager:
## ldap_rootdn: "dc=example,dc=com"
##
## Password of LDAP manager:
## ldap_password: "******"
##
## Search base of LDAP directory:
## ldap_base: "dc=example,dc=com"
##
## LDAP attribute that holds user ID:
## ldap_uids:
##   - "mail": "%u@mail.example.org"
##
## LDAP filter:
## ldap_filter: "(objectClass=shadowAccount)"

##
## Anonymous login support:
##   auth_method: anonymous
##   anonymous_protocol: sasl_anon | login_anon | both
##   allow_multiple_connections: true | false
##
## host_config:
##   "public.example.org":
##     auth_method: anonymous
##     allow_multiple_connections: false
##     anonymous_protocol: sasl_anon
##
## To use both anonymous and internal authentication:
##
## host_config:
##   "public.example.org":
##     auth_method:
##       - internal
##       - anonymous

###.  ==============
###'  DATABASE SETUP

## ejabberd by default uses the internal Mnesia database,
## so you do not necessarily need this section.
## This section provides configuration examples in case
## you want to use other database backends.
## Please consult the ejabberd Guide for details on database creation.

##
## MySQL server:
##
sql_type: mysql
sql_server: "xxxxxx"
sql_database: "xxxxx"
sql_username: "xxxx"
sql_password: "xxxxx"
sql_port: xxxxx

##
## If you want to specify the port:
## sql_port: 1234

##
## PostgreSQL server:
##
## sql_type: pgsql
## sql_server: "server"
## sql_database: "database"
## sql_username: "username"
## sql_password: "password"
##
## If you want to specify the port:
## sql_port: 1234
##
## If you use PostgreSQL, have a large database, and need a
## faster but inexact replacement for "select count(*) from users"
##
## pgsql_users_number_estimate: true

##
## SQLite:
##
## sql_type: sqlite
## sql_database: "/path/to/database.db"

##
## ODBC compatible or MSSQL server:
##
## sql_type: odbc
## sql_server: "DSN=ejabberd;UID=ejabberd;PWD=ejabberd"

##
## Number of connections to open to the database for each virtual host
##
## sql_pool_size: 10

##
## Interval to make a dummy SQL request to keep the connections to the
## database alive. Specify in seconds: for example 28800 means 8 hours
##
## sql_keepalive_interval: undefined

###.  ===============
###'  TRAFFIC SHAPERS

shaper:
  ##
  ## The "normal" shaper limits traffic speed to 1000 B/s
  ##
  normal: 1000

  ##
  ## The "fast" shaper limits traffic speed to 50000 B/s
  ##
  fast: 50000

##
## This option specifies the maximum number of elements in the queue
## of the FSM. Refer to the documentation for details.
##
max_fsm_queue: 10000

###.   ====================
###'   ACCESS CONTROL LISTS
acl:
  ##
  ## The 'admin' ACL grants administrative privileges to XMPP accounts.
  ## You can put here as many accounts as you want.
  ##
  admin:
     user:
       - "xxxx"
  ##     - "ermine@example.org"
  ##
  ## Blocked users
  ##
  ## blocked:
  ##   user:
  ##     - "baduser@example.org"
  ##     - "test"

  ## Local users: don't modify this.
  ##
  local:
    user_regexp: ""

  ##
  ## More examples of ACLs
  ##
  ## jabberorg:
  ##   server:
  ##     - "jabber.org"
  ## aleksey:
  ##   user:
  ##     - "aleksey@jabber.ru"
  ## test:
  ##   user_regexp: "^test"
  ##   user_glob: "test*"

  ##
  ## Loopback network
  ##
  loopback:
    ip:
      - "127.0.0.0/8"
      - "::1/128"
      - "::FFFF:127.0.0.1/128"

  ##
  ## Bad XMPP servers
  ##
  ## bad_servers:
  ##   server:
  ##     - "xmpp.zombie.org"
  ##     - "xmpp.spam.com"

##
## Define specific ACLs in a virtual host.
##
## host_config:
##   "localhost":
##     acl:
##       admin:
##         user:
##           - "bob-local@localhost"

###.  ============
###'  SHAPER RULES

shaper_rules:
  ## Maximum number of simultaneous sessions allowed for a single user:
  max_user_sessions: 10
  ## Maximum number of offline messages that users can have:
  max_user_offline_messages:
    - 5000: admin
    - 100
  ## For C2S connections, all users except admins use the "normal" shaper
  c2s_shaper:
    - none: admin
    - normal
  ## All S2S connections use the "fast" shaper
  s2s_shaper: fast

###.  ============
###'  ACCESS RULES
access_rules:
  ## This rule allows access only for local users:
  local:
    - allow: local
  ## Only non-blocked users can use c2s connections:
  c2s:
    - deny: blocked
    - allow
  ## Only admins can send announcement messages:
  announce:
    - allow: admin
  ## Only admins can use the configuration interface:
  configure:
    - allow: admin
  ## Only accounts of the local ejabberd server can create rooms:
  muc_create:
    - allow: local
  ## Only accounts on the local ejabberd server can create Pubsub nodes:
  pubsub_createnode:
    - allow: local
  ## In-band registration allows registration of any possible username.
  ## To disable in-band registration, replace 'allow' with 'deny'.
  register:
    - allow
  ## Only allow to register from localhost
  trusted_network:
    - allow: loopback
  ## Do not establish S2S connections with bad servers
  ## If you enable this you also have to uncomment "s2s_access: s2s"
  ## s2s:
  ##   - deny:
  ##     - ip: "XXX.XXX.XXX.XXX/32"
  ##   - deny:
  ##     - ip: "XXX.XXX.XXX.XXX/32"
  ##   - allow

## ===============
## API PERMISSIONS
## ===============
##
## This section allows you to define who and using what method
## can execute commands offered by ejabberd.
##
## By default "console commands" section allow executing all commands
## issued using ejabberdctl command, and "admin access" section allows
## users in admin acl that connect from 127.0.0.1 to  execute all
## commands except start and stop with any available access method
## (ejabberdctl, http-api, xmlrpc depending what is enabled on server).
##
## If you remove "console commands" there will be one added by
## default allowing executing all commands, but if you just change
## permissions in it, version from config file will be used instead
## of default one.
##
api_permissions:
  "console commands":
    from:
      - ejabberd_ctl
    who: all
    what: "*"
  "admin access":
    who:
      - access:
          - allow:
            - acl: loopback
            - acl: admin
      - oauth:
        - scope: "ejabberd:admin"
        - access:
          - allow:
            - acl: loopback
            - acl: admin
    what:
      - "*"
      - "!stop"
      - "!start"
  "public commands":
    who:
      - ip: "127.0.0.1/8"
    what:
      - "status"
      - "connected_users_number"

## By default the frequency of account registrations from the same IP
## is limited to 1 account every 10 minutes. To disable, specify: infinity
## registration_timeout: 600
  
##
## Define specific Access Rules in a virtual host.
##
## host_config:
##   "localhost":
##     access:
##       c2s:
##         - allow: admin
##         - deny
##       register:
##         - deny

###.  ================
###'  DEFAULT LANGUAGE

##
## language: Default language used for server messages.
##
language: "en"

##
## Set a different default language in a virtual host.
##
## host_config:
##   "localhost":
##     language: "ru"

###.  =======
###'  CAPTCHA

##
## Full path to a script that generates the image.
##
## captcha_cmd: "/lib/ejabberd/priv/bin/captcha.sh"

##
## Host for the URL and port where ejabberd listens for CAPTCHA requests.
##
## captcha_host: "example.org:5280"

##
## Limit CAPTCHA calls per minute for JID/IP to avoid DoS.
##
## captcha_limit: 5

###.  ====
###'  ACME
##
## In order to use the acme certificate acquiring through "Let's Encrypt"
## an http listener has to be configured to listen to port 80 so that
## the authorization challenges posed by "Let's Encrypt" can be solved.
## 
## A simple way of doing this would be to add the following in the listening
## section and to configure port forwarding from 80 to 5280 either via NAT
## (for ipv4 only) or using frontends such as haproxy/nginx/sslh/etc.
##   - 
##    port: 5280
##    ip: "::"
##    module: ejabberd_http

acme:

   ## A contact mail that the ACME Certificate Authority can contact in case of
   ## an authorization issue, such as a server-initiated certificate revocation.
   ## It is not mandatory to provide an email address but it is highly suggested.
   contact: "mailto:example-admin@example.com"


   ## The ACME Certificate Authority URL.
   ## This could either be:
   ##   - https://acme-v01.api.letsencrypt.org - (Default) for the production CA
   ##   - https://acme-staging.api.letsencrypt.org - for the staging CA
   ##   - http://localhost:4000 - for a local version of the CA
   ca_url: "https://acme-v01.api.letsencrypt.org"

###.  =======
###'  MODULES

##
## Modules enabled in all ejabberd virtual hosts.
##
modules:
  mod_adhoc: {}
  mod_admin_extra: {}
  mod_announce: # recommends mod_adhoc
    access: announce
  mod_blocking: {} # requires mod_privacy
  mod_caps: {}
  mod_carboncopy: 
    default: always
  mod_client_state: {}
  mod_configure: {} # requires mod_adhoc
  ## mod_delegation: {} # for xep0356
  mod_disco: {}
  mod_echo: {}
  mod_irc: {}
  mod_bosh: {}
  ## mod_http_fileserver:
  ##   docroot: "/var/www"
  ##   accesslog: "/var/log/ejabberd/access.log"
  ## mod_http_upload:
  ##   # docroot: "@HOME@/upload"
  ##   put_url: "https://@HOST@:5444"
  ##   thumbnail: false # otherwise needs the identify command from ImageMagick installed
  ## mod_http_upload_quota:
  ##   max_days: 30
  mod_last: {}
  ## XEP-0313: Message Archive Management
  ## You might want to setup a SQL backend for MAM because the mnesia database is
  ## limited to 2GB which might be exceeded on large servers
  mod_mam: 
    default: always
    db_type: sql # for xep0313, mnesia is limited to 2GB, better use an SQL backend
  mod_muc:
    ## host: "conference.@HOST@"
    access:
      - allow
    access_admin:
      - allow: admin
    access_create: muc_create
    access_persistent: muc_create
  mod_muc_admin: {}
  ## mod_muc_log: {}
  ## mod_multicast: {}
  mod_offline:
    access_max_user_messages: max_user_offline_messages
  mod_ping: {}
  ## mod_pres_counter:
  ##   count: 5
  ##   interval: 60
  mod_privacy: {}
  mod_private: {}
  ## mod_proxy65: {}
  mod_pubsub:
    access_createnode: pubsub_createnode
    ## reduces resource comsumption, but XEP incompliant
    ignore_pep_from_offline: true
    ## XEP compliant, but increases resource comsumption
    ## ignore_pep_from_offline: false
    last_item_cache: false
    plugins:
      - "flat"
      - "hometree"
      - "pep" # pep requires mod_caps
  mod_push: {}
  mod_push_keepalive: {}
  ## mod_register:
    ##
    ## Protect In-Band account registrations with CAPTCHA.
    ##
    ##   captcha_protected: true
    ##
    ## Set the minimum informational entropy for passwords.
    ##
    ##   password_strength: 32
    ##
    ## After successful registration, the user receives
    ## a message with this subject and body.
    ##
    ##   welcome_message:
    ##     subject: "Welcome!"
    ##     body: |-
    ##       Hi.
    ##       Welcome to this XMPP server.
    ##
    ## When a user registers, send a notification to
    ## these XMPP accounts.
    ##
    ##   registration_watchers:
    ##     - "admin1@example.org"
    ##
    ## Only clients in the server machine can register accounts
    ##
    ##   ip_access: trusted_network
    ##
    ## Local c2s or remote s2s users cannot register accounts
    ##
    ##   access_from: deny
    ##   access: register
  mod_roster: {}
  mod_shared_roster: {}
  mod_stats: {}
  mod_time: {}
  mod_vcard:
    search: false
  mod_vcard_xupdate: {}
  ## Convert all avatars posted by Android clients from WebP to JPEG
  ## mod_avatar:  # this module needs compile option --enable-graphics
  ##   convert:
  ##     webp: jpeg
  mod_version: {}
  mod_stream_mgmt: {}
  ##   Non-SASL Authentication (XEP-0078) is now disabled by default
  ##   because it's obsoleted and is used mostly by abandoned
  ##   client software
  ## mod_legacy_auth: {}
  ##   The module for S2S dialback (XEP-0220). Please note that you cannot
  ##   rely solely on dialback if you want to federate with other servers,
  ##   because a lot of servers have dialback disabled and instead rely on
  ##   PKIX authentication. Make sure you have proper certificates installed
  ##   and check your accessibility at https://check.messaging.one/
  mod_s2s_dialback: {}
  mod_http_api: {}

##
## Enable modules with custom options in a specific virtual host
##
## host_config:
##   "localhost":
##     modules:
##       mod_echo:
##         host: "mirror.localhost"

##
## Enable modules management via ejabberdctl for installation and
## uninstallation of public/private contributed modules
## (enabled by default)
##

allow_contrib_modules: true

###.
###'
### Local Variables:
### mode: yaml
### End:
### vim: set filetype=yaml tabstop=8 foldmarker=###',###. foldmethod=marker:
```
My android application code
```
 void connect() throws IOException, XMPPException, SmackException, InterruptedException {
        SmackConfiguration.DEBUG = true;
        lastMessages = mInboxDatabase.getLastMessages();
        Log.d(TAG, "Connecting to server " + ChatConnectionService.SERVICE_NAME);
        XMPPTCPConnectionConfiguration.Builder conf = XMPPTCPConnectionConfiguration.builder()
                .setXmppDomain(ChatConnectionService.SERVICE_NAME)
                .setKeystorePath(null)
                .setKeystoreType(null)
                .setPort(5222)
                .setResource("androidDevb")
                .setSecurityMode(ConnectionConfiguration.SecurityMode.disabled)
                .setCompressionEnabled(true)
                .setDebuggerEnabled(true);

        Log.d(TAG, "Username : " + mUsername);
        Log.d(TAG, "Password : " + mPassword);
        Log.d(TAG, "Server : " + ChatConnectionService.SERVICE_NAME);
        mConnection = new XMPPTCPConnection(conf.build());
        XMPPTCPConnection.setUseStreamManagementDefault(true);
        SASLAuthentication.unBlacklistSASLMechanism("PLAIN");
        SASLAuthentication.blacklistSASLMechanism("X-OAUTH2");
        // connect to Xmpp server
        mConnection.addConnectionListener(this);
        initIncomingMessageListener();
        try {
            Log.d(TAG, "Calling connect() ");
            mConnection.connect();
            Log.d(TAG, " login() Called ");
            mConnection.login(mUsername, mPassword);
            sLoggedInState = LoggedInState.LOGGED_IN;
        } catch (Exception e) {
            e.printStackTrace();
        }
        initCarbonMessageManager();
        initReconnectionManager();
    }
```

> was the unexpected behavior? What was the expected result?

Authentication response from my python script.

**Comment by nosnilmot** [comment]
Your username contains uppercase character(s), same issue as #2280, PR #2446 fixes it. Workaround for now would be to specify username in lowercase (which I believe Adium is doing for you).

**Comment by weiss** [comment]
Fixed in 4f8af723c6fc5dc4b689f0f1734d4e2b11dd4c82.
